src/core/src/service_interfaces/StatusHandler.py

```diff
@@ -798,17 +844,213 @@ def __try_add_error(error_list, detail):
         error_list.insert(0, detail)
         return True
 
-    def __set_errors_json(self, error_count_by_operation, errors_by_operation):
+    def __set_errors_json(self, error_count_by_operation, errors_by_operation, truncated=False):
         """ Compose the error object json to be added in 'errors' in given operation's summary """
-        message = "{0} error/s reported.".format(error_count_by_operation)
-        message += " The latest {0} error/s are shared in detail. To view all errors, review this log file on the machine: {1}".format(len(errors_by_operation), self.__log_file_path) if error_count_by_operation > 0 else ""
+        success_code = Constants.PatchOperationTopLevelErrorCode.SUCCESS
+        error_code = Constants.PatchOperationTopLevelErrorCode.ERROR
+        error_count = error_count_by_operation
+        code = success_code if error_count_by_operation == 0 else error_code
+
+        # Update the errors json to include truncation detail
+        if truncated:
+            error_count += 1    # add 1 because of truncation
+            code = Constants.PatchOperationTopLevelErrorCode.WARNING if code != error_code else error_code
+
+        message = "{0} error/s reported.".format(error_count)
+        message += " The latest {0} error/s are shared in detail. To view all errors, review this log file on the machine: {1}".format(len(errors_by_operation), self.__log_file_path) if error_count > 0 else ""
         return {
-            "code": Constants.PatchOperationTopLevelErrorCode.SUCCESS if error_count_by_operation == 0 else Constants.PatchOperationTopLevelErrorCode.ERROR,
+            "code": code,
             "details": errors_by_operation,
             "message": message
         }
+
+    def __set_error_detail(self, error_code, message):
+        formatted_message = self.__ensure_error_message_restriction_compliance(message)
+        return {
+            "code": str(error_code),
+            "message": str(formatted_message)
+        }
     # endregion
 
+    # region - Patch Truncation
+    def log_truncated_packages(self):
+        """ log the removed packages from patches in CoreMain after main operation are marked completed """
+        if not len(self.__assessment_packages_removed) == 0:
+            self.composite_logger.log_debug("Packages removed from assessment packages list: {0}".format(self.__assessment_packages_removed))
+        if not len(self.__installation_packages_removed) == 0:
```

Comment:
No test coverage

Reply:
done. it is difficult to just test a method that do just log and no return without using mock. maybe in the future, we return a boolean value to make the test easier?